services/src/main/java/io/scalecube/services/ServiceCall.java

```diff
@@ -322,15 +322,14 @@ public ServiceCall create() {
   }
 
   private Mono<Address> addressLookup(ServiceMessage request) {
-    return router
-        .route(serviceRegistry, request)
-        .map(serviceReference -> Mono.just(serviceReference.address()))
-        .orElseGet(() -> Mono.error(noReachableMemberException(request)))
-        .doOnError(
-            t -> {
-              Optional<Object> dataOptional = Optional.ofNullable(request.data());
-              dataOptional.ifPresent(ReferenceCountUtil::safestRelease);
-            });
+    return Mono.fromCallable(() -> router.route(serviceRegistry, request)
+      .map(ServiceReference::address)
+      .orElseThrow(() -> noReachableMemberException(request)))
+      .doOnError(
+        t -> {
+          Optional<Object> dataOptional = Optional.ofNullable(request.data());
+          dataOptional.ifPresent(ReferenceCountUtil::safestRelease);
```

Comment:
Would you mind if I ask you to inline `dataOptional` and geti rid of curly braces block.